File: 1679.cpp
#include<iostream>
#include<algorithm>
#include<vector>

using namespace std;

class Solution{
public:
    int maxOperations(vector<int>& nums, int k){
        sort(nums.begin(), nums.end());
        int i = 0, j = nums.size() - 1;
        int ans = 0;
        while(i < j){
            if(nums[i] + nums[j] == k){
                ans++;
                i++;
                j--;
            }
            else if(nums[i] + nums[j] < k) i++;
            else j--;
        }
        return ans;
    }
};

int main(){
    int n;
    cin >> n;
    vector<int> nums(n);
    for(int i=0; i < n; i++) 
        cin >> nums[i];
    int k;
    cin >> k;
    Solution s;
    cout << s.maxOperations(nums, k) << endl;
    return 0;
}

File: python/1679.py
class Solution:
    def maxOperations(self, nums: list[int], k: int) -> int:
        nums.sort()
        i, j, ans = 0, len(nums) - 1, 0
        while i < j:
            if nums[i] + nums[j] == k:
                ans += 1
                i += 1
                j -= 1
            elif nums[i] + nums[j] < k:
                i += 1
            else:
                j -= 1
        return ans

if __name__ == '__main__':
    nums = list(map(int, input().split()))
    k = int(input())
    s = Solution()
    print(s.maxOperations(nums, k))